src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -1172,8 +1173,8 @@ public void AcceptNewConnection(Stream stream, Action disconnect, EndPoint remot
                     switch (initialMessage.InitialMessageType)
                     {
                         case InitialMessageType.RequestVote:
-                            var elector = new Elector(this, remoteConnection);
-                            elector.Run();
+                            using (var elector = new Elector(this, remoteConnection))
+                                elector.RunAndWait();
```

Comment:
No, it is a separate one, this is why I'm blocking.

Reply:
I believe that we have this stack:

* Raven.Server.ListenToNewTcpConnection - start a new task, which is in a loop
* RavenServer.DispatchTcpConnection
* RavenServer.DispatchServerWideTcpConnection
* ServerStore.ClusterAcceptNewConnection
* RachisConsensus.AcceptNewConnection
* Elector.Run

There are four of them (`ListenToNewTcpConnection` tasks), so you are blocking a very critical limited resource that you are blocking here.

Reply:
I think we are spawning a new one before dispatching https://github.com/ravendb/ravendb/blob/7e1121ef6bf2b4f5fd518145ed51745441cba013/src/Raven.Server/RavenServer.cs#L1881 ?

Reply:
You are correct, I though it was a `while`, but it isn't.
What is the reasoning behind wanting to block here? 

I still don't like it (convoy on the thread pool issue)

Reply:
Previously we were not waiting anywhere for the Elector at all. I cannot use subscribe to the current disposal mechanism because it relies on order of subscribers and Electors were never taken into account there. Separate mechanism could be added just for electors, but @karmeli87 was not a fan of this.

Reply:
We can maybe limit one (or two electors) per source? 
e.g. Node A can have a single elector talking to B at a time.

Reply:
I don't like that idea. Electors needs to be able to run independently, otherwise we may have them stuck and unable to progress.